charger/daheimladen.go

```diff
@@ -0,0 +1,193 @@
+	cc := struct {
+		Token     string
+		StationID string
+		IDTag     string
+	}{}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	return NewDaheimLaden(cc.Token, cc.StationID, cc.IDTag)
+}
+
+// NewDaheimLaden creates DaheimLaden charger
+func NewDaheimLaden(token string, stationID string, idTag string) (*DaheimLaden, error) {
+	c := &DaheimLaden{
+		Helper:      request.NewHelper(util.NewLogger("daheim")),
+		stationID:   stationID,
+		connectorID: 1,
+		idTag:       idTag,
+		token:       token,
+	}
+	c.Timeout = 15
+	c.Client.Transport = &oauth2.Transport{
+		Source: oauth2.StaticTokenSource(&oauth2.Token{
+			AccessToken: token,
+			TokenType:   "Bearer",
+		}),
+		Base: c.Client.Transport,
+	}
+	return c, nil
+}
+
+// Enabled implements the api.Charger interface
+func (c *DaheimLaden) Enabled() (bool, error) {
+	var res daheimladen.GetLatestStatus
+	err := c.GetJSON(fmt.Sprintf("%s/cs/%s/status", daheimladen.BASE_URL, c.stationID), &res)
+	return res.Status == string(daheimladen.CHARGING) || res.Status == string(daheimladen.PREPARING), err
+}
+
+// Enable implements the api.Charger interface
+func (c *DaheimLaden) Enable(enable bool) error {
+	if enable {
+		data := daheimladen.RemoteStartRequest{
+			ConnectorID: c.connectorID,
+			IdTag:       c.idTag,
+		}
+
+		uri := fmt.Sprintf("%s/cs/%s/remotestart", daheimladen.BASE_URL, c.stationID)
+		req, err := http.NewRequest(http.MethodPost, uri, request.MarshalJSON(data))
+		if err != nil {
+			return err
+		}
+
+		var res daheimladen.RemoteStartResponse
+		if err = c.DoJSON(req, &res); err == nil && res.Status != string(daheimladen.REMOTE_START_ACCEPTED) {
+			err = fmt.Errorf("charging station refused to start transaction")
+		}
+
+		return err
+	}
+
+	var res daheimladen.GetLatestInProgressTransactionResponse
+	uri := fmt.Sprintf("%s/cs/%s/get_latest_inprogress_transaction", daheimladen.BASE_URL, c.stationID)
+	if err := c.GetJSON(uri, &res); err != nil {
+		return err
+	}
+
+	c.transactionID = res.TransactionID
+	if c.transactionID == 0 {
+		return fmt.Errorf("cannot stop transaction as the transaction was started with plug and charge mode")
+	}
+
+	data := daheimladen.RemoteStopRequest{
+		TransactionID: c.transactionID,
+	}
+
+	uri = fmt.Sprintf("%s/cs/%s/remotestop", daheimladen.BASE_URL, c.stationID)
+	req, err := http.NewRequest(http.MethodPost, uri, request.MarshalJSON(data))
+	if err != nil {
+		return err
+	}
+
+	var remoteStopRes daheimladen.RemoteStartResponse
+	if err = c.DoJSON(req, &remoteStopRes); err == nil && remoteStopRes.Status != string(daheimladen.REMOTE_STOP_ACCEPTED) {
+		err = fmt.Errorf("charging station refused to stop transaction")
+	}
+
+	return err
+}
+
+// MaxCurrent implements the api.Charger interface
+func (c *DaheimLaden) MaxCurrent(current int64) error {
+	data := daheimladen.ChangeConfigurationRequest{
+		Key:   string(daheimladen.CHARGE_RATE),
+		Value: fmt.Sprint(current),
```

Comment:
Should be strconv.Itoa?

Reply:
Done

Reply:
I'm using `strconv.FormatInt(i int64, base int)` as `current` is of type int64 and the docs say `strconv.Iota` is just the equivalent of `strconv.FormatInt(int64(i), 10)`